src/coreComponents/finiteVolume/FluxApproximationBase.hpp

```diff
@@ -245,10 +248,10 @@ class FluxApproximationBase : public dataRepository::Group
   string m_coeffName;
 
   /// names of coefficient models to build the stencil for
-  string_array m_coefficientModelNames;
+  map< string, array1d< string > > m_coefficientModelNames; // TODO: remove
```

Comment:
I think the only part of this class that still uses both `coefficientName` and `coefficientModelNames` is the boundary stencil, which has not been updated to use the new permeability model and still assumes constant values. If we still think that having face-based Dirichlet boundary conditions for flow is valuable, it would be great to have a separate PR to update that kernel (it's going to be somewhat tricky because the kernel already uses fluid dispatch to compute fluid properties at the face, adding permeability dispatch will make it quite nasty).

Reply:
Ok thanks, I see. To me having the faced-based Dirichlet boundary conditions is very valuable, so I agree with this plan of updating the face-based BC kernel in a separate PR.